specs/Standards/Tokens/NonFungibleToken.md

```diff
@@ -0,0 +1,196 @@
+
+#### **Technical calls**
+
+1. `alice` makes an async call to `corgi::grantAccess({"escrow_account_id":"escrow", "token_id":3})`
+2. `jerry`  makes an async call to ``sausage::grantAccess({"escrow_account_id":"escrow", "token_id":5})`
+3. In one Promise
+    1. `escrow` calls `sausage::transferFrom({"owner_id":"jerry", "new_owner_id:"alice", "token_id": 5})`
+        1. attaches callback `escrow::onTransfer({"owner_id":"jerry", "token_contract":"sausage", "token_id": 5})`
+    2. `escrow` calls `corgi::transferFrom({"owner_id":"alice", "new_owner_id:"jerry", "token_id": 3})`
+        1. attaches callback `escrow::onTransfer({"owner_id":"alice", "token_contract:"corgi", "token_id": 3})`
+
+
+#### Example of pseudocode for escrow transfer
+
+```TypeScript
+[...]// [Corgi Contract]
+corgiToken.grantAccess(corgiTokenId, escrowID);
+
+[...] // [Sausage Contract]
+sausageContract.grantAccess(sausageTokenId, escrowID);
+
+[...] // [Escrow Contract]
+async escrowContractCall() {
+  await Promise.all([
+    corgiToken.transferFrom(tokenId, aliceID, jerryID);
+    sausageContract.transferFrom(tokenId, jerryID, aliceID);
+  ]);
+}
+
+```
+
+## Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+## Template for smart contract in AssemblyScript
+
+
+At time of writing, this standard is established with several constraints found in AssemblyScript. The first is that interfaces are not an implemented feature of AssemblyScript, and the second is that classes are not exported in the conversion from AssemblyScript to WASM. This means that the entire contract could be implemented as a class, which might be better for code organization, but it would be deceiving in function.
+
+```TypeScript
+  type TokenId = u64;
+
+  /******************/
+  /* CHANGE METHODS */
+  /******************/
+  
+  // Grant the access to the given `accountId` for the given `tokenId`.
+  // Requirements:
+  // * The caller of the function (`predecessor_id`) should have access to the token.
+  export function grantAccess(tokenId: TokenId, accountId: string): void;
+
+  // Revoke the access to the given `accountId` for the given `tokenId`.
+  // Requirements:
+  // * The caller of the function (`predecessor_id`) should have access to the token.
+  export function revokeAccess(tokenId: TokenId, accountId: string): void;
+
+  // Transfer the given `tokenId` from the given `accountId`.  Account `newAccountId` becomes the new owner.
+  // Requirements:
+  // * The caller of the function (`predecessor_id`) should have access to the token.
+  export function transferFrom(tokenId: TokenId, accountId: string, newAccountID: string): void;
+
+
+  // Transfer the given `tokenId` to the given `accountId`.  Account `accountId` becomes the new owner.
+  // The token unlocks (if it was locked) and all access is revoked except for the new owner.
```

Comment:
If you don't grant access per token, then you don't need to change `access` for the transferred token, which makes implementation simpler.